rewrite-maven/src/main/java/org/openrewrite/maven/AddAnnotationProcessor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.XmlIsoVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Optional;
+
+import static org.openrewrite.xml.AddOrUpdateChild.addOrUpdateChild;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddAnnotationProcessor extends Recipe {
+    private static final XPathMatcher PLUGINS_MATCHER = new XPathMatcher("/project/build/plugins");
+    private static final String MAVEN_COMPILER_PLUGIN_GROUP_ID = "org.apache.maven.plugins";
+    private static final String MAVEN_COMPILER_PLUGIN_ARTIFACT_ID = "maven-compiler-plugin";
+
+    @Option(displayName = "Group",
+            description = "The first part of the coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION' of the processor to add.",
+            example = "org.projectlombok")
+    String groupId;
+
+    @Option(displayName = "Artifact",
+            description = "The second part of a coordinate 'org.openrewrite.maven:rewrite-maven-plugin:VERSION' of the processor to add.",
+            example = "lombok-mapstruct-binding")
+    String artifactId;
+
+    @Override
+    public String getDisplayName() {
+        return "Change Maven plugin configuration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Apply the specified configuration to a Maven plugin. Will not add the plugin if it does not already exist in the pom.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenVisitor<ExecutionContext>() {
+            @Override
+            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag plugins = (Xml.Tag) super.visitTag(tag, ctx);
+                if (PLUGINS_MATCHER.matches(getCursor())) {
```

Comment:
Did you consider using the Maven plugin trait here? Or was that not an option?
https://github.com/openrewrite/rewrite/blob/7eb376f178d0a3930124e8e72ad2b8441190e393/rewrite-maven/src/test/java/org/openrewrite/maven/trait/Traits.java#L26-L28